src/fp2.rs

```diff
@@ -553,29 +568,29 @@ impl Fp2 {
     }
 
     pub fn invert(&self) -> CtOption<Self> {
-        // #[cfg(target_os = "zkvm")]
-        // {
-        //     // Compute the inverse using the zkvm syscall
-        //     unconstrained! {
-        //         let mut buf = [0u8; 97];
-        //         self._invert().map(|inv| {
-        //             buf[..96].copy_from_slice(&inv.to_bytes());
-        //             buf[96] = 1;
-        //         });
-        //         hint_slice(&buf);
-        //     }
+        #[cfg(target_os = "zkvm")]
+        {
+            // Compute the inverse using the zkvm syscall
+            unconstrained! {
+                let mut buf = [0u8; 97];
+                self._invert().map(|inv| {
+                    buf[..96].copy_from_slice(&inv.to_bytes());
+                    buf[96] = 1;
+                });
+                hint_slice(&buf);
+            }
 
-        //     let byte_vec = read_vec();
-        //     let bytes: [u8; 97] = byte_vec.try_into().unwrap();
-        //     match bytes[96] {
-        //         0 => CtOption::new(Fp2::zero(), Choice::from(0u8)),
-        //         _ => {
-        //             let inv = Fp2::from_bytes(&bytes[0..96].try_into().unwrap()).unwrap();
-        //             CtOption::new(inv, !self.is_zero() & (self * inv).ct_eq(&Fp2::one()))
-        //         }
-        //     }
-        // }
-        // #[cfg(not(target_os = "zkvm"))]
+            let byte_vec = read_vec();
+            let bytes: [u8; 97] = byte_vec.try_into().unwrap();
+            match bytes[96] {
+                0 => CtOption::new(Fp2::zero(), Choice::from(0u8)),
+                _ => {
+                    let inv = Fp2::from_bytes(&bytes[0..96].try_into().unwrap()).unwrap();
+                    CtOption::new(inv, !self.is_zero() & (self * inv).ct_eq(&Fp2::one()))
+                }
+            }
+        }
```

Comment:
Add a more detailed comment for why you're doing `_invert`. Might be better to call it CPU `invert`.